rust/agents/relayer/src/relayer.rs

```diff
@@ -166,21 +182,29 @@ impl Relayer {
         .instrument(info_span!("run inbox"))
     }
 
-    pub fn run(&self) -> Instrumented<JoinHandle<Result<()>>> {
+    pub async fn run(&self) -> Instrumented<JoinHandle<Result<()>>> {
         let (signed_checkpoint_sender, signed_checkpoint_receiver) =
-            tokio::sync::watch::channel::<Option<MultisigSignedCheckpoint>>(None);
-
-        let mut tasks: Vec<Instrumented<JoinHandle<Result<()>>>> = self
-            .inboxes()
-            .iter()
-            .map(|(inbox_name, inbox_contracts)| {
-                self.run_inbox(
-                    inbox_contracts.clone(),
-                    signed_checkpoint_receiver.clone(),
-                    self.core.settings.inboxes[inbox_name].gelato_conf.clone(),
-                )
-            })
-            .collect();
+            watch::channel::<Option<MultisigSignedCheckpoint>>(None);
+
+        let inboxes = self.inboxes();
+
+        let mut tasks: Vec<Instrumented<JoinHandle<Result<()>>>> =
```

Comment:
Not just this line but some around here as well, is there any real benefit to specifying this type?